### PR TITLE
Node Annotation Update: CoreNodeModels MapCombineLacing DesignScriptBuiltIn.Dictionary

### PR DESCRIPTION
**Pull request by martinstacey**
### Purpose
The following PR addresses nodes' names and descriptions to make them more descriptive for beginners, PRs are organized according to location in DynamoDS.

These corrections are for CoreNodeModels.MapCombineLacing and  DesignScriptBuiltIn.Dictionary

**NODE 177 : ReplaceByCondition**
Description Input:  Predicate used to determine if it should be replaced      ->        Replacement function used to determine if an object should be replaced								
Description Output:  If condition(item) = True, then replaceWith is returend. Otherwise item is returned unaltered      ->        Item, or list with items replaced by condition	

**NODE 183 : List.Filter**									
Description Input:  Predicate used to determine if an element is filtered in or out.      ->        Filter function used to determine if an element is filtered in our out									

**NODE 129 : Dictionary.Components**					
Description Output:  The keys of the dictionary      ->        Keys of the dictionary	
Description Output:  The values of the dictionary      ->        Values of the dictionary							

**NODE 344 : Dictionary.Values**								
Description Output:  The values of the Dictionary.      ->        Values of the dictionary									

**NODE 345 : Dictionary.Keys**
Description Output:  The keys of the Dictionary      ->        Keys of the dictionary									

**NODE 346 : Dictionary.ValueAtKey**				
Description Input:  The key in the Dictionary to obtain.      ->        The key in the dictionary to obtain value				
Description Output:  The value at the specified key or null if it is not set.      ->        Value at the specified key or null if it is not set									

**NODE 347 : Dictionary.SetValueAtKeys**								
Description Output:  A new Dictionary with the entries inserted      ->        New dictionary with the entries inserted									

**NODE 530 : Dictionary.Count**		
Description Output:  int      ->        Number of key value pairs in dictionary									

**NODE 531 : Dictionary.RemoveKeys**				
Description Input:  The key in the Dictionary to remove      ->        Keys to remove			
Description Output:  A new Dictionary with the key removed      ->        New dictionary with keys removed							

**NODE 532 : Dictionary.ByKeysValues**
Description Input:  The string keys of the Dictionary      ->        Keys of dictionary
Description Input:  The values of the Dictionary      ->        Values of dictionary							
Description Output:  The result Dictionary      ->        Dictionary from keys and values

### FYI
For nodes Dictionary.Components, Dictionary.Values, Dictionary.Keys, Dictionary.ValueAtKey, Dictionary.SetValueAtKeys, Dictionary.Count, Dictionary.RemoveKeys, Dictionary.ByKeysValues These nodes are using a property of class Dictionary, so I do not know how to change the description for input location. I would make these changes to the inputs descriptions:
Description Input:  DesignScript.Builtin.Dictionary      ->        Dictionary to extract components from	
Description Input:  DesignScript.Builtin.Dictionary      ->        Dictionary to query values from	
Description Input:  DesignScript.Builtin.Dictionary      ->        Dictionary to query value at key from
Description Input:  DesignScript.Builtin.Dictionary      ->        Dictionary to set value at key	
Description Input:  DesignScript.Builtin.Dictionary      ->        Dictionary to query keys from
Description Input:  DesignScript.Builtin.Dictionary      ->        Dictionary to count number of key value pairs from
Description Input:  DesignScript.Builtin.Dictionary      ->        Dictionary to remove keys from

image

### Reviewers
@Amoursol @aparajit-pratap @QilongTang

### CODE REVIEW

File: src/Libraries/DesignScriptBuiltin/Dictionary.cs [hunk 1]
```diff
@@ -98,8 +98,8 @@ public Dictionary RemoveKeys(IList<string> keys)
             /// <summary>
             ///     Obtain the value at a specified key
             /// </summary>
-            /// <param name="key">The key in the Dictionary to obtain.</param>
-            /// <returns name="value">The value at the specified key or null if it is not set.</returns>
+            /// <param name="key">The key in the Dictionary to obtain value from</param>
```
Comment (aparajit-pratap) [inline src/Libraries/DesignScriptBuiltin/Dictionary.cs:101]: `...obtain value for.`

Comment (martinstacey) [reply src/Libraries/DesignScriptBuiltin/Dictionary.cs]: Addressed in 30f8ef5

File: src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx [hunk 2]
```diff
@@ -226,7 +226,7 @@
     <comment>Description for File Path</comment>
   </data>
   <data name="FilterPortDataConditionToolTip" xml:space="preserve">
-    <value>Predicate used to determine if an element is filtered in or out.</value>
+    <value>Filter function used to determine if an element is filtered in our out</value>
```
Comment (mmisol) [inline src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx:229]: Typo: `our` => `or`

File: src/Libraries/CoreNodeModels/Properties/Resources.resx [hunk 3]
```diff
@@ -226,7 +226,7 @@
     <comment>Description for File Path</comment>
   </data>
   <data name="FilterPortDataConditionToolTip" xml:space="preserve">
-    <value>Predicate used to determine if an element is filtered in or out.</value>
+    <value>Filter function used to determine if an element is filtered in our out</value>
```
Comment (mmisol) [inline src/Libraries/CoreNodeModels/Properties/Resources.resx:229]: Same here

File: src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx [hunk 4]
```diff
@@ -435,7 +435,7 @@
     <value>Object to replace with</value>
   </data>
   <data name="ReplacePortDataResultToolTip" xml:space="preserve">
-    <value>If condition(item) = True, then "replaceWith" is returned. Otherwise "item" is returned unaltered.</value>
+    <value>Item, or list with items replaced by condition</value>
```
Comment (mmisol) [inline src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx:438]: The wording is strange here. The items are not replaced by a condition, but rather they are conditionally replaced.

File: src/Libraries/CoreNodeModels/Properties/Resources.resx [hunk 5]
```diff
@@ -435,7 +435,7 @@
     <value>Object to replace with</value>
   </data>
   <data name="ReplacePortDataResultToolTip" xml:space="preserve">
-    <value>If condition(item) = True, then "replaceWith" is returned. Otherwise "item" is returned unaltered.</value>
+    <value>Item, or list with items replaced by condition</value>
```
Comment (mmisol) [inline src/Libraries/CoreNodeModels/Properties/Resources.resx:438]: Same here.

**Review by mmisol — COMMENTED** [review]
@martinstacey Added some minor comments. Please take a look. Thanks!

File: src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx (hunk 2 above)
Comment (martinstacey) [reply src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx]: Addressed in a4098fd

File: src/Libraries/CoreNodeModels/Properties/Resources.resx (hunk 3 above)
Comment (martinstacey) [reply src/Libraries/CoreNodeModels/Properties/Resources.resx]: Addressed in a4098fd

File: src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx (hunk 4 above)
Comment (martinstacey) [reply src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx]: Addressed in a4098fd

File: src/Libraries/CoreNodeModels/Properties/Resources.resx (hunk 5 above)
Comment (martinstacey) [reply src/Libraries/CoreNodeModels/Properties/Resources.resx]: > Same here.

Addressed in a4098fd

**Comment by martinstacey** [comment]
> @martinstacey Added some minor comments. Please take a look. Thanks!

Thanks @mmisol! will have a look at the comments.

**Review by mmisol — APPROVED** [review]
Thanks for addressing the changes @martinstacey . Looks good to me

**Pull request merged**